src/wp-includes/blocks.php

```diff
@@ -1344,6 +1344,69 @@ function traverse_and_serialize_block( $block, $pre_callback = null, $post_callb
 	);
 }
 
+/**
+ * Replaces patterns in a block tree with their content.
+ *
+ * @since 6.6.0
+ *
+ * @param array $blocks An array blocks.
+ *
+ * @return array An array of blocks with patterns replaced by their content.
```

Comment:
From what I know, there’s no need to add an empty line between the `@param` and `@return` tags. Also, I noticed the description for the `@inner_content` function argument is missing. Could that be added?
```suggestion
 * @param array $blocks         An array blocks.
 * @param array $inner_content  An optional array of inner content.
 * @return array An array of blocks with patterns replaced by their content.
```

Reply:
This is meant to be a private arg, not sure if there's a better way https://github.com/WordPress/wordpress-develop/pull/6673/files#r1623854851

Reply:
There might be a solution to get rif of the $inner_content parameter:
```php
/**
 * Replaces patterns in a block tree with their content.
 *
 * @since 6.6.0
 *
 * @param array $blocks An array of blocks of a parent block.
 * @return array An array of blocks with patterns replaced by their content.
 */
function resolve_pattern_blocks( $blocks ) {
	// Keep track of seen references to avoid infinite loops.
	static $seen_refs = array();
	$i                = 0;
	$is_parent_block = ! empty( $blocks['innerBlocks'] );

	if ( $is_parent_block ) {
		// If it's a single block, $blocks has to be reinitialized before it can be used.
		$block         = $blocks;
		$blocks        = $block['innerBlocks'];
		$inner_content = $block['innerContent'];
	}
	while ( $i < count( $blocks ) ) {
		if ( 'core/pattern' !== $blocks[ $i ]['blockName'] ) {
			if ( $is_parent_block ) {
				$blocks[ $i ] = resolve_pattern_blocks( $blocks[ $i ] );
			}
			++$i;
			continue;
		}
		$attrs = $blocks[ $i ]['attrs'];

		if ( empty( $attrs['slug'] ) ) {
			++$i;
			continue;
		}

		$slug = $attrs['slug'];

		if ( isset( $seen_refs[ $slug ] ) ) {
			// Skip recursive patterns.
			array_splice( $blocks, $i, 1 );
			continue;
		}

		$registry = WP_Block_Patterns_Registry::get_instance();
		$pattern  = $registry->get_registered( $slug );

		// Skip unknown patterns.
		if ( ! $pattern ) {
			++$i;
			continue;
		}

		$blocks_to_insert   = parse_blocks( $pattern['content'] );
		$seen_refs[ $slug ] = true;
		$blocks_to_insert   = resolve_pattern_blocks( $blocks_to_insert );
		unset( $seen_refs[ $slug ] );
		array_splice( $blocks, $i, 1, $blocks_to_insert );

		// If we have inner content, we need to insert nulls in the
		// inner content array, otherwise serialize_blocks will skip
		// blocks.
		if ( !empty( $inner_content ) ) {
			$null_indices  = array_keys( $inner_content, null, true );
			$content_index = $null_indices[ $i ];
			$nulls         = array_fill( 0, count( $blocks_to_insert ), null );
			array_splice( $inner_content, $content_index, 1, $nulls );
		}

		// Skip inserted blocks.
		$i += count( $blocks_to_insert );
	}

	if ( $is_parent_block ) {
		$block['innerBlocks']  = $blocks;
		$block['innerContent'] = $inner_content;
		return $block;
	}

	return $blocks;
}
```

Reply:
Tbh, I find it a bit more confusing. Also, why is it now sometimes returning a blocks array and sometimes one block?

Reply:
> Tbh, I find it a bit more confusing.

I feel the same way.

> why is it now sometimes returning a blocks array and sometimes one block?

I made this change to eliminate the `$inner_content` parameter, aiming to streamline the function's interface.
I've tried to refactor this function into two separate functions, but it didn't go well either. This likely indicates that the current structure  might be the most efficient approach.